packages/hub/routes/wyre-callback.ts

```diff
@@ -0,0 +1,349 @@
+import Koa from 'koa';
+import autoBind from 'auto-bind';
+import Logger from '@cardstack/logger';
+import DatabaseManager from '../services/database-manager';
+import { inject } from '../di/dependency-injection';
+import WyreService, { WyreOrder, WyreTransfer, WyreWallet } from '../services/wyre';
+import Web3 from 'web3';
+
+const { toChecksumAddress } = Web3.utils;
+let log = Logger('route:wyre-callback');
+
+interface WyreCallbackRequest {
+  id: string;
+  source: string;
+  dest: string;
+  currency: string;
+  amount: number;
+  status: WyreTransfer['status'];
+  createdAt: number;
+}
+
+interface ValidatedWalletReceiveRequest {
+  order: WyreOrder;
+  transfer: WyreTransfer;
+  wallet: WyreWallet;
+}
+interface ValidatedWalletSendRequest {
+  orderId: string;
+  reservationId: string | null;
+  userAddress: string;
+  transfer: WyreTransfer;
+}
+
+export const adminWalletName = 'admin';
+
+export default class WyreCallbackRoute {
+  adminWalletId: string | undefined;
+  wyre: WyreService = inject('wyre');
+  prepaidCardInventory = inject('prepaid-card-inventory', { as: 'prepaidCardInventory' });
+  databaseManager: DatabaseManager = inject('database-manager', { as: 'databaseManager' });
+
+  constructor() {
+    autoBind(this);
+  }
+
+  async post(ctx: Koa.Context) {
+    let request = ctx.request.body;
+    log.info(`Received wyre callback: ${JSON.stringify(request, null, 2)}`);
```

Comment:
Are there any security issues to consider in logging this request body?

Reply:
No, there is no personally identifiable info, no names, no credit card numbers, no addresses. Just wyre generated identifiers, and transaction amounts.